### PR TITLE
Add external link tracking to Fathom integration

### PR DESCRIPTION
**Pull request by KevinRohn**
This PR adds an option to track external link clicks in the Fathom integration. Regarding to the issue #450 

When enabled, the integration tracks clicks on links leading to external domains. It sends an event to Fathom with the format "External link clicked: domain".

- Modified `index.ts` to pass the new option to the script
- Updated `gitbook-manifest.yaml` with the new configuration
- Updated `fathomScript.raw.js` implement external link tracking
- Small cosmetic variable change `siteID` renamed to `siteId`, this naming is also for other integrations

I hope I did anything correct, because I was not able to test it without the script injection scope.

Cheers,
Kevin

### CODE REVIEW

**Comment by addisonschultz** [comment]
Changes look good. But before I merge this, can you install and test this version to verify it works on your end too? 

 https://app.gitbook.com/integrations/test-fathom-update
 
 If all is good, I can get this merged in, and deployed to the production version :)

**Comment by KevinRohn** [comment]
Hey @addisonschultz , the tracking it self works, but I do not get a event for external site clicks. So it does not work yet.

**Comment by KevinRohn** [comment]
Can I simply update this PR and it will reflect the changes in my space?

**Comment by addisonschultz** [comment]
You can update this PR - and I'll update the integration after!

**Comment by KevinRohn** [comment]
Hi @addisonschultz,

I made a mistake in the replace function call...
I fixed it with the last commit. See screenshot. 
Screenshot 2024-09-13 at 17 07 20

Can you please update the integration?

**Comment by addisonschultz** [comment]
It should be updated now - you can install with the same link! https://app.gitbook.com/test-fathom-update

**Comment by KevinRohn** [comment]
@addisonschultz please update again.

**Comment by addisonschultz** [comment]
Done ✅ The integration looks good - is it working as expected on your end as well?

**Comment by KevinRohn** [comment]
Hi @addisonschultz,

now it works as expected!

File: integrations/fathom/gitbook-manifest.yaml [hunk 1]
```diff
@@ -36,11 +36,27 @@ summary: |
 categories:
     - analytics
 configurations:
+    space:
```
Comment (addisonschultz) [inline integrations/fathom/gitbook-manifest.yaml:39]: We don't use this `space` configuration key anymore - so you can omit this one from the configuration (See `main` right now, as it's been removed

File: integrations/fathom/src/index.ts [hunk 2]
```diff
@@ -26,7 +27,16 @@ export const handleFetchEvent: FetchPublishScriptEventCallback = async (
         return;
     }
 
-    return new Response(script.replace('<TO_REPLACE>', siteId), {
+    const trackExternalLinks =
+        environment.siteInstallation?.configuration?.track_external_links ??
+        environment.spaceInstallation?.configuration?.track_external_links ??
```
Comment (addisonschultz) [inline integrations/fathom/src/index.ts:32]: You can omit `environment.spaceInstallation...` from here as well

**Review by addisonschultz — CHANGES_REQUESTED** [review]
If you could make these changes, I'll be happy to merge and get this all live on the production version 😄

**Comment by KevinRohn** [comment]
> If you could make these changes, I'll be happy to merge and get this all live on the production version 😄

Sure, I will do this.

Please keep this PR open - I want to provide some changes to the event information. 
```js
window.fathom.trackEvent('External link clicked: ' + domainName);
```

The `domainName` info is not enough.

File: integrations/fathom/src/index.ts (hunk 2 above)
Comment (KevinRohn) [reply integrations/fathom/src/index.ts]: Done

File: integrations/fathom/gitbook-manifest.yaml (hunk 1 above)
Comment (KevinRohn) [reply integrations/fathom/gitbook-manifest.yaml]: Done

**Comment by KevinRohn** [comment]
@addisonschultz ok, please update again for a final test

**Comment by addisonschultz** [comment]
Sorry for the delay! It's just been updated again

**Comment by KevinRohn** [comment]
Screenshot 2024-09-18 at 10 24 36

Can you please check the `gitbook-manifest.yaml`. After making the desired changes, I can no longer use the configuration page?

**Comment by addisonschultz** [comment]
Looking into it :/

**Comment by addisonschultz** [comment]
@KevinRohn can you give it a try now? We should have fixed the bug causing this!

**Comment by KevinRohn** [comment]
@addisonschultz Thank you so much for your quick fix and prompt reply.

I've had the time to test the implementation in detail and discovered an issue with the previous version:

Links present when the site initially loaded were being tracked correctly. However, links that appeared after navigating between pages (due to dynamic content loading) were not being tracked. This occurred because the original script didn't account for dynamically added links.

Fix:
I've implemented an event listener that now keeps track of all links, including those loaded dynamically after the initial page load. This ensures tracking across the entire site, regardless of when content is loaded.

Please re-deploy for a final test :) 

Thank you in advance

**Comment by addisonschultz** [comment]
Latest version deployed 😄

**Comment by KevinRohn** [comment]
> @KevinRohn can you give it a try now? We should have fixed the bug causing this!

@addisonschultz sorry that I have to contact you again with something else.

Is it possible that the previous bug fix has created a new bug?
I have “uninstalled” the integration as a test. However, the script still remains on the page?

**Comment by KevinRohn** [comment]
@addisonschultz

OK, I understand now. The space and site configuration for the integration was a bit confusing for me at first. I've tested my latest change, and it works as expected. So, feel free to merge it.

Screen Recording 2024-09-23 at 13 09 06

**Comment by KevinRohn** [comment]
> @KevinRohn looks good - one last thing! Can you add a changeset `npm run changeset` in the root of the integrations folder, and add a changeset? you can bump this a minor version, so it will get deployed to the main version after 😄
> 
> Thanks!

Done ✅

**Review by addisonschultz — APPROVED** [review]

**Pull request merged**